factory/usr/lib/the-modeenv

```diff
@@ -20,4 +20,3 @@ if grep -q snapd_recovery_mode=run /proc/cmdline; then
 fi
 mount -o bind /run/image.fstab /sysroot/etc/fstab
 umount -lf /sysroot/run
-cp /run/systemd/transient/*.mount /etc/systemd/system
```

Comment:
Because it copies to `/etc` which is only the ramfs, I believe this has never done anything. If it was copied to a bind mount within `/sysroot/etc`, maybe...

Reply:
During `initrd.target` `systemctl daemon-reload` is executed, and at that point, inside the initrd the units from `/run/systemd/transient/` were getting lost as unreferenced.

I think today, snap-bootstrap always generates/copies units in both `/run` and `/etc` such that they survive `daemon-reload` prior to `pivot-root`, thus it is possible that this `cp` call is now redundant.

overall if things still boot correctly, and the mount points are all there, and not recreated, things are good.

Reply:
Are you sure they survived the switch root? Files in `/etc` should not survive. It is files in `/sysroot/etc` that will survive.

Reply:
This did not survive switch root, but they did survive daemon-reload inside the initrd.

Reply:
Ah, that makes more sense. Though, I am surprised `daemon-reload` loses track of them. It sounds like a bug. I do not see where systemd filters transient units when serializing. I will do some testing.

We still have one `daemon-reload` done by `initrd-parse-etc.service` from systemd. I will see if that loses state.

Reply:
systemd doesn't really filter transient units, it just `rm -rf` /run/systemd and then reruns generators, which are expected to recreate all the things. Ideally, i was hoping that somehow `systemd-mount` had an option to create mount units persistent i.e. in /etc, instead of in `/run/` but it doesn't really support this.

Reply:
I have tested it, `systemctl daemon-reload` does not remove files in `/run/systemd/transient`, and all the mounts are still listed in `systemctl` (active and mounted). Could it be an older version of systemd that did that?